app/assets/javascripts/tsheets_integration.js

```diff
@@ -0,0 +1,65 @@
+var tsheets = (function($) {
```

Comment:
Thanks! As for js I don't like the way we're just encouraged to dump everything in `/javascripts`, and all code is run on every page. For projects with not very complex javascript I quite like to use the Garber Irish pattern for executing javascript based on controller and action (https://www.viget.com/articles/extending-paul-irishs-comprehensive-dom-ready-execution), but I'm wondering if it might be worth going for react in this project - I know @creaumond is writing react native in another project, so would already be familiar with it, what's everyone's thoughts?